src/coreclr/jit/lower.cpp

```diff
@@ -1270,20 +1266,31 @@ bool Lowering::TryLowerSwitchToBitTest(
         comp->fgAddRefPred(bbCase1, bbSwitch);
     }
 
+    var_types bitTableType = (bitCount <= (genTypeSize(TYP_INT) * 8)) ? TYP_INT : TYP_LONG;
+    GenTree*  bitTableIcon = comp->gtNewIconNode(bitTable, bitTableType);
+
+#ifdef TARGET_XARCH
     //
     // Append BT(bitTable, switchValue) and JCC(condition) to the switch block.
     //
-
-    var_types bitTableType = (bitCount <= (genTypeSize(TYP_INT) * 8)) ? TYP_INT : TYP_LONG;
-    GenTree*  bitTableIcon = comp->gtNewIconNode(bitTable, bitTableType);
-    GenTree*  bitTest      = comp->gtNewOperNode(GT_BT, TYP_VOID, bitTableIcon, switchValue);
+    GenTree* bitTest = comp->gtNewOperNode(GT_BT, TYP_VOID, bitTableIcon, switchValue);
     bitTest->gtFlags |= GTF_SET_FLAGS;
     GenTreeCC* jcc = comp->gtNewCC(GT_JCC, TYP_VOID, bbSwitchCondition);
-
     LIR::AsRange(bbSwitch).InsertAfter(switchValue, bitTableIcon, bitTest, jcc);
-
+#else  // TARGET_XARCH
+    //
+    // Fallback to AND(RSZ(bitTable, switchValue), 1)
+    //
+    GenTree* tstCns = comp->gtNewIconNode(bbSwitch->bbNext != bbCase0 ? 0 : 1, bitTableType);
+    GenTree* shift  = comp->gtNewOperNode(GT_RSZ, bitTableType, bitTableIcon, switchValue);
+    GenTree* one    = comp->gtNewIconNode(1, bitTableType);
+    GenTree* andOp  = comp->gtNewOperNode(GT_AND, bitTableType, shift, one);
+    GenTree* cmp    = comp->gtNewOperNode(GT_EQ, TYP_INT, andOp, tstCns);
+    GenTree* jcc    = comp->gtNewOperNode(GT_JTRUE, TYP_VOID, cmp);
+    LIR::AsRange(bbSwitch).InsertAfter(switchValue, bitTableIcon, shift, tstCns, one);
+    LIR::AsRange(bbSwitch).InsertAfter(one, andOp, cmp, jcc);
+#endif // !TARGET_XARCH
```

Comment:
Why do we still need the xarch special case? Does `OptimzeConstCompare` not handle the case? Ideally we would teach it about the missing opportunity so that everyone benefits instead of special casing it here

Reply:
I thought about it, but I'd like to leave it for a future follow up since it involves more work to add that peephole and I was mainly interesting in improving arm64

Reply:
Ah, you mean that the peephole already exists?

Reply:
Ok, sounds fine to me. I would personally expect `x & (1 << y)` to be quite common, and it looks like we are missing this opportunity for arm64 (`TEST_EQ/TEST_NE(x, LSH(1, y)) => TEST_EQ/TEST_NE(RSZ(x, y), 1)`). Then this transform could produce `EQ/NE(AND(x, LSH(1, y)))`

Reply:
I don't think it exists, but it seems like adding it here would be straightforward:
https://github.com/dotnet/runtime/blob/e04c75e7c65814f7f4813262effbd56d2c90e5e3/src/coreclr/jit/lower.cpp#L3596-L3621

Reply:
Of course we could also teach it about `TEST_EQ/TEST_NE(RSZ(x, y), 1) => BT(x, y)` on x64/x86, but I guess this is a less common pattern

Reply:
Anyway I'm ok with keeping this PR as is.

Reply:
@jakobbotsch ok let's then keep as is and I'll file a "good-first-issue" to recognize BT pattern (or will work myself when I have time)